### PR TITLE
refactor: Remove bindenv from settings & centralize settings creation

### PR DESCRIPTION
**Pull request by KevLehman**

### CODE REVIEW

**Comment by KevLehman** [comment]
Drafting for now as meteor still doesn't support top level await on 3.0

File: apps/meteor/app/lib/server/startup/oAuthServicesUpdate.js [hunk 1]
```diff
@@ -194,10 +194,10 @@ function customOAuthServicesInit() {
...
-				addOAuthService(name, values);
+				await addOAuthService(name, values);
 			}
 		}
-	});
+	}
 }
 
-customOAuthServicesInit();
+await customOAuthServicesInit();
```
Comment (MarcosSpessatto) [inline apps/meteor/app/lib/server/startup/oAuthServicesUpdate.js:203]: Would not be better to create a function as you did in the other ones? 🤔

File: apps/meteor/server/settings/accounts.ts [hunk 2]
```diff
@@ -0,0 +1,949 @@
...
+			await this.add('Accounts_Password_History_Amount', 5, {
+				type: 'int',
+				enableQuery,
+				i18nLabel: 'Password_History_Amount',
+				i18nDescription: 'Password_History_Amount_Description',
+			});
+		});
+	});
+
+await settingsRegistry.addGroup('OAuth', async function () {
```
Comment (MarcosSpessatto) [inline apps/meteor/server/settings/accounts.ts:794]: Any reason not to have this inside a function as well?

**Review by MarcosSpessatto — DISMISSED** [review]
I've found these other 2 cases [orchestrator.ts](https://github.com/RocketChat/Rocket.Chat/blob/refactor/bind-env-settings/apps/meteor/ee/server/apps/orchestrator.js#L243-L293) and [assets.ts](https://github.com/RocketChat/Rocket.Chat/blob/refactor/bind-env-settings/apps/meteor/app/assets/server/assets.ts#L359-L364). Shouldn't we move those 2 to functions as well?

File: apps/meteor/app/lib/server/startup/oAuthServicesUpdate.js (hunk 1 above)
Comment (KevLehman) [reply apps/meteor/app/lib/server/startup/oAuthServicesUpdate.js:203]: Ah, this is already a function 😬 but didn't move to the other places as this has some extra logic in there (getting settings, env, creating oauth classes etc) so left it for a future refactor: separate setting creation logic from oauth logic and break down the file

File: apps/meteor/server/settings/accounts.ts (hunk 2 above)
Comment (KevLehman) [reply apps/meteor/server/settings/accounts.ts:794]: Good catch! This should be on Oauth file 😬

**Comment by KevLehman** [comment]
About assets and orchestrator:

- orchestrator is EE so left intact (since we have the UI and BE to decide whether or not to use these, I don't think it affects but yeah, ee)
- assets has some "customization" to the settings, so left intact

**Review by rodrigok — APPROVED** [review]

**Pull request merged**